### ISSUE TITLE
Model Creation Bug When Using an Entity with a Name Less than 4 Characters

### ISSUE THREAD

**Issue by jjbutera** [msallin/SQLiteCodeFirst#66]
I get the following error because I have an entity called "Set" (like a music set) in my model.

`StartIndex cannot be less than zero. Parameter name: startIndex
   at System.String.Remove(Int32 startIndex, Int32 count)
   at SQLite.CodeFirst.Utility.SqliteAssociationType.IsSelfReferencing(AssociationType associationType)
   at SQLite.CodeFirst.Utility.SqliteAssociationType..ctor(AssociationType associationType, EntityContainer container)
...`

This code throws the exception. I would check the ToRole.Name length first. If less than 4, return false;

```
private const string SelfReferencingPostfix = "Self";
private static bool IsSelfReferencing(AssociationType associationType)
{
            return associationType.Constraint.ToRole.Name.Remove(associationType.Constraint.ToRole.Name.Length - SelfReferencingPostfix.Length, SelfReferencingPostfix.Length) == associationType.Constraint.FromRole.Name;
}
```

Maybe change to:

```
private static bool IsSelfReferencing(AssociationType associationType)
{
    var toRoleName = associationType.Constraint.ToRole.Name;
    if (toRoleName.Length < 4)
        return false;

    return toRoleName.Remove(toRoleName.Length - SelfReferencingPostfix.Length, SelfReferencingPostfix.Length) == associationType.Constraint.FromRole.Name;
}
```

**Comment by jjbutera** [comment]
It looks like you don't like entity names less than 4 characters. It doesn't have anything to do with a keyword. I tried another short entity name with same results. I can see where the problem is...shorter than the postfix "Self"

```
private const string SelfReferencingPostfix = "Self";

private static bool IsSelfReferencing(AssociationType associationType)
        {
            return associationType.Constraint.ToRole.Name.Remove(associationType.Constraint.ToRole.Name.Length - SelfReferencingPostfix.Length, SelfReferencingPostfix.Length) == associationType.Constraint.FromRole.Name;
        }
```

**Comment by msallin** [comment]
Hello @jjbutera 

Thank you for reporting the bug, the analysis and the suggestion to solve this issue.
I just implemented the fix. For details please have a look at the [PR](https://github.com/msallin/SQLiteCodeFirst/pull/67).

The new version, which includes this fix, is 1.2.3. Grap it from [NuGet](https://www.nuget.org/packages/SQLite.CodeFirst/) or download it from [GitHub Release](https://github.com/msallin/SQLiteCodeFirst/releases).

Regards Marc
